tests/test_authz.py

```diff
@@ -220,3 +223,39 @@ async def test_resolve_async_resolver(test_authz: AuthorizationExtension):
         await test_authz.resolve(async_resolver, None, None)
         == "Hello, World! From async resolver"
     )
+
+
+@pytest.mark.asyncio
+async def test_wrapped_extension(test_authz: AuthorizationExtension):
+    test_authz.set_required_global_permissions(["read:Comments"])
+    decorated_resolver = test_authz.require_permissions(
+        [], ignore_global_permissions=True
+    )(resolver)
+
+    async def next_extension(*args, **kwargs):
+        return decorated_resolver(*args, **kwargs)
+
+    with pytest.raises(GraphQLErrorAuthorizationError):
+        await test_authz.resolve(next_extension, None, None)
```

Comment:
Failure case is illustrated here when the registered resolver cannot be found and the input to `test_authz.resolve` is not the decorated_resolver which is an instance of `PermissionChecker`.